spatial/r3/transform.go

```diff
@@ -0,0 +1,267 @@
+// Copyright ©2022 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package r3
+
+import (
+	"math"
+)
+
+// Transform represents a 3D spatial transformation.
+// The zero value of Transform is the identity transform.
+type Transform struct {
```

Comment:
What about making this structured like the `Mat` type rather than having distinct fields for each element?

Reply:
Right now this PR makes no assumption about the underlying data type and is pretty fast, maybe owing to the SSA optimizations made possible by having each element be a field of the struct. The changes you suggest would require making substantial changes. I'd like to leave this to a future PR that 
* demonstrates that an array representation does not substantially slow down the implementation
* Maintains the zero value Transform is the identity transform

Reply:
SGTM